src/conf/domain_conf.c

```diff
@@ -29795,6 +29801,12 @@ virDomainDefFormatInternalSetRootName(virDomainDefPtr def,
         virDomainLoaderDefFormat(buf, def->os.loader);
     virBufferEscapeString(buf, "<kernel>%s</kernel>\n",
                           def->os.kernel);
+    virBufferEscapeString(buf, "<snp>%s</snp>\n",
```

Comment:
Instead of introducing new element 'snp' here, please try to extend existing SEV support. 
https://libvirt.org/kbase/launch_security_sev.html#amd-secure-encrypted-virtualization-sev

has more details on how to use AMD SEV in libvirt.